b64.h

```diff
@@ -28,50 +28,55 @@ char *b64Encode(char *data)
         'w', 'x', 'y', 'z', '0', '1', '2', '3',
         '4', '5', '6', '7', '8', '9', '+', '/'};
 
-    ssize_t inlen = strlen(data);
+    int external_flag = 0;
+    if (dst != NULL)
+        external_flag++;
+
+    ssize_t inlen = srclen;
     ssize_t outlen = B64_LEN(inlen);
 
-    char *out = malloc(outlen + 1);
-    if (out == NULL) return NULL;
+    char *out = (external_flag) ? dst : malloc(outlen + 1);
+    if (out == NULL)
+        return NULL;
     out[outlen] = '\0';
     char *p = out;
 
     ssize_t i;
     for (i = 0; i < inlen - 2; i += 3)
     {
-        *p++ = b64e[(data[i] >> 2) & 0x3F];
-        *p++ = b64e[((data[i] & 0x3) << 4) | ((data[i + 1] & 0xF0) >> 4)];
-        *p++ = b64e[((data[i + 1] & 0xF) << 2) | ((data[i + 2] & 0xC0) >> 6)];
-        *p++ = b64e[data[i + 2] & 0x3F];
+        *p++ = b64e[(src[i] >> 2) & 0x3F];
+        *p++ = b64e[((src[i] & 0x3) << 4) | ((src[i + 1] & 0xF0) >> 4)];
+        *p++ = b64e[((src[i + 1] & 0xF) << 2) | ((src[i + 2] & 0xC0) >> 6)];
+        *p++ = b64e[src[i + 2] & 0x3F];
     }
 
     if (i < inlen)
     {
-        *p++ = b64e[(data[i] >> 2) & 0x3F];
+        *p++ = b64e[(src[i] >> 2) & 0x3F];
         if (i == (inlen - 1))
         {
-            *p++ = b64e[((data[i] & 0x3) << 4)];
+            *p++ = b64e[((src[i] & 0x3) << 4)];
             *p++ = '=';
         }
         else
         {
-            *p++ = b64e[((data[i] & 0x3) << 4) | ((data[i + 1] & 0xF0) >> 4)];
-            *p++ = b64e[((data[i + 1] & 0xF) << 2)];
+            *p++ = b64e[((src[i] & 0x3) << 4) | ((src[i + 1] & 0xF0) >> 4)];
+            *p++ = b64e[((src[i + 1] & 0xF) << 2)];
         }
         *p++ = '=';
     }
 
     return out;
 }
 
-char *b64Decode(char *data)
+char *b64Decode(void *dst, char *src, size_t srclen)
```

Comment:
The decode is going from a string to a bytes, so if you want you can assume src is a null terminated string you could remove `srclen`.